### ISSUE TITLE
Different build behaviour for .net core web apps between local VS IDE and msbuild

### ISSUE THREAD

**Issue by fromorbonia** [dotnet/cli#10556]
## Steps to reproduce
Publishing a .NET Core web application with a FileSystem profile under VS2017 IDE creates a diretory with an EXE (e.g. MyProject.exe)
There is a Jenkins build server running msbuild, which correctly picks up the publish profile that was used and configured locally.

## Expected  behavior
I would expect the build server to also create exactly the same set of files (specifically in this case it should include MyProject.exe).

## Actual behavior
The publish folder contains an apphost.exe and MyProject.dll, but no EXE. 
The web.config file created actually references MyProject.exe (in processPath), and of course the web site doesn't not run on IIS.

## Environment data
`dotnet --info` output:
For the local VS2017 IDE this is:
.NET Core SDK (reflecting any global.json):
 Version:   2.1.401
 Commit:    91b1c13032

Runtime Environment:
 OS Name:     Windows
 OS Version:  6.1.7601
 OS Platform: Windows
 RID:         win7-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.401\

Host (useful for support):
  Version: 2.1.3
  Commit:  124038c13e

.NET Core SDKs installed:
  2.1.101 [C:\Program Files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.401 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

For the Jenkins build server:
12:53:26 .NET Core SDK (reflecting any global.json):
12:53:26  Version:   2.1.502
12:53:26  Commit:    c74ce8f29f
12:53:26 
12:53:26 Runtime Environment:
12:53:26  OS Name:     Windows
12:53:26  OS Version:  10.0.14393
12:53:26  OS Platform: Windows
12:53:26  RID:         win10-x64
12:53:26  Base Path:   C:\Program Files\dotnet\sdk\2.1.502\
12:53:26 
12:53:26 Host (useful for support):
12:53:26   Version: 2.1.6
12:53:26   Commit:  3f4f8eebd8
12:53:26 
12:53:26 .NET Core SDKs installed:
12:53:26   2.1.401 [C:\Program Files\dotnet\sdk]
12:53:26   2.1.500 [C:\Program Files\dotnet\sdk]
12:53:26   2.1.502 [C:\Program Files\dotnet\sdk]
12:53:26 
12:53:26 .NET Core runtimes installed:
12:53:26   Microsoft.AspNetCore.All 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
12:53:26   Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
12:53:26   Microsoft.AspNetCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
12:53:26   Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
12:53:26   Microsoft.NETCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
12:53:26   Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

**Comment by fromorbonia** [comment]
Just thinking it might be better to post this issue in msbuild area, but dotnet publish behaves in a similar way.

**Comment by livarcocc** [comment]
Are you passing the a rid to the publish command or msbuild operation? In VS, it is generating the exe for you because you are specifying a RID there, which implies to the publish operation that you are building a self-contained app.

For msbuild or dotnet publish, if you pass that rid there as well, you should see the same results.

**Comment by fromorbonia** [comment]
Hi @livarcocc - no I wasn't explicity setting the RID. However, it was in the Publish Profile ("win-x64"), so that should have been applied in both situations?

Just to confirm that I ran on the build server a build with /p:RuntimeIdentifiers="win-x64" and then with /p:RuntimeIdentifiers="win7-x64" - no change in the files created for publishing.

Independant of that, I would also say that doesn't explain the inconsitency with the web.config file. If the build process has decided to create an apphost.exe, surely the web.config file should contain settings for that?

**Comment by livarcocc** [comment]
In order to be treated as a self-contained app, you need to pass RuntimeIdentifier, singular.

**Comment by livarcocc** [comment]
As for the web.config, I would file an issue at aspnet/websdk. That's is generated there.

**Comment by fromorbonia** [comment]
@livarcocc - thanks for the useful replies and the pointer to aspnet/websdk, will post there.

FYI - sorry should have said I wasn't aiming at self contained. When I first got the profile working it only had (in the pubxml file):
<_IsPortable>true</_IsPortable>

Over successive attempts I ended up with:
    <UseAppHost>true</UseAppHost>
    <SelfContained>false</SelfContained>
    <_IsPortable>true</_IsPortable>
(Although on the local VS IDE only IsPortable = true was needed to publish a web site that folder that worked correctly when placed on the IIS server).
